includes/Utilities/Tracker.php

```diff
@@ -45,22 +73,95 @@ public function add_tracker_data( array $data = array() ) {
 			$data['extensions'] = array();
 		}
 
-		// Is the site connected?
-		// @since 2.3.4
+		/**
+		 * Is the site connected?
+		 *
+		 * @since 2.3.4
+		 */
 		$connection_is_happy = false;
 		$connection_handler  = facebook_for_woocommerce()->get_connection_handler();
 		if ( $connection_handler ) {
 			$connection_is_happy = $connection_handler->is_connected() && ! get_transient( 'wc_facebook_connection_invalid' );
 		}
 		$data['extensions']['facebook-for-woocommerce']['is-connected'] = wc_bool_to_string( $connection_is_happy );
 
-		// What features are enabled on this site?
-		// @since 2.4.0
+		/**
+		 * What features are enabled on this site?
+		 *
+		 * @since 2.3.4
+		 */
 		$product_sync_enabled = facebook_for_woocommerce()->get_integration()->is_product_sync_enabled();
 		$data['extensions']['facebook-for-woocommerce']['product-sync-enabled'] = wc_bool_to_string( $product_sync_enabled );
 		$messenger_enabled = facebook_for_woocommerce()->get_integration()->is_messenger_enabled();
 		$data['extensions']['facebook-for-woocommerce']['messenger-enabled'] = wc_bool_to_string( $messenger_enabled );
 
+		/**
+		 * Has the feed file been requested recently?
+		 *
+		 * @since x.x.x
+		 */
+		$feed_file_requested = get_transient( self::TRANSIENT_WCTRACKER_FEED_REQUESTED );
```

Comment:
You need to protect the code from expired transients:

```bash
[01-Jun-2021 10:54:28 UTC] PHP Notice:  Trying to get property 'ig_cta_enabled' of non-object in /Users/bartoszbudzanowski/Local/facebook/app/public/wp-content/plugins/facebook-for-woocommerce/includes/Utilities/Tracker.php on line 114
```

Reply:
I think I've fixed this here: d9d57beac9b814c0a2c33bfe1ac839cd296fb819

I couldn't reproduce the notice though - not sure why. I have `WP_DEBUG` set to true, and I deleted the transient, then actually forced `$config = false`. Let me know if there's some other debugging settings I need to enable.